src/v2/models/AppState.js

```diff
@@ -101,55 +103,50 @@ export default Model.extend({
     return currentViewState;
   },
 
-  setIonResponse (idxResp) {
+  setIonResponse (transformedResponse) {
     // Don't re-render view if new response is same as last.
     // Usually happening at polling and pipeline doesn't proceed to next step.
     // expiresAt will be different for each response, hence compare objects without that property
-    if (_.isEqual(_.omit(idxResp.rawIdxState, 'expiresAt'), _.omit(this.get('__previousResponse'), 'expiresAt'))) {
+    const previousRawState = this.has('idx') ? this.get('idx').rawIdxState : null;
+    if (_.isEqual(_.omit(transformedResponse.idx.rawIdxState, 'expiresAt'), _.omit(previousRawState, 'expiresAt') )) {
       return;
     }
 
     // `currentFormName` is default to first form of remediation object or nothing.
-    idxResp.currentFormName = null;
+    transformedResponse.currentFormName = null;
```

Comment:
this would going to trigger `render` and later on one of the if condition can also again trigger `render` because now we listen for `change:currentFormName` event ? can we suppress this double event trigger ? or is it required ?

Reply:
why `transformedResponse.currentFormName = null` will trigger `render` event? 
It doesn't set to Model yet.

Reply:
ah okay make sense.